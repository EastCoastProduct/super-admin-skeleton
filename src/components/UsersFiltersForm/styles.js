import { StyleSheet } from 'aphrodite/no-important';

const shared = {
  inputs: {
    display: 'inline-block',
    verticalAlign: 'middle',
  },
};

export default StyleSheet.create({
  checkbox: {
    ...shared.inputs,
    marginRight: 40,
  },
  searchBar: {
    ...shared.inputs,
    width: 350,
  },
});
