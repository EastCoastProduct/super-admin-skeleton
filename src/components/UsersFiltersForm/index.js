import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form/immutable';
import { css } from 'aphrodite/no-important';
import Checkbox from '../Checkbox';
import SearchBar from '../SearchBar';
import styles from './styles';

export function UsersFiltersForm(props) {
  const { className, form, handleFiltersSubmit, handleSubmit } = props;

  return (
    <form
      className={css(className && className)}
      onSubmit={handleSubmit(handleFiltersSubmit)}
      noValidate
    >
      <Field
        className={styles.checkbox}
        name="confirmed"
        component={Checkbox}
        id={form}
        onChange={handleSubmit(handleFiltersSubmit)}
        label="Show only confirmed users"
      />
      <Field
        className={styles.searchBar}
        name="search"
        component={SearchBar}
        placeholder="Search by email, first and last name"
        normalize={value => `%${value}%`}
        format={value => (value ? value.slice(1, -1) : '')}
      />
    </form>
  );
}

UsersFiltersForm.propTypes = {
  className: PropTypes.object,
  form: PropTypes.string.isRequired,
  handleFiltersSubmit: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'UsersFilters',
})(UsersFiltersForm);
