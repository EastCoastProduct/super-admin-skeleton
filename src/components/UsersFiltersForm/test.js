import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import UsersFiltersForm from './';

jest.mock('redux-form/immutable', () => ({
  Field: 'Field',
  reduxForm: () => component => component,
}));

describe('UsersFiltersForm component', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    jest.resetAllMocks(); // move to before when clearAllMocks gets out
  });

  const wrapper = shallow(
    <UsersFiltersForm
      form="Form"
      handleFiltersSubmit={jest.fn()}
      handleSubmit={jest.fn(cb => cb())}
    />
  );
  const instance = wrapper.instance();
  const searchBar = wrapper.find('[name="search"]')

  it('renders and submits on submit', () => {
    wrapper.find('form').simulate('submit');

    expect(instance.props.handleSubmit)
      .toHaveBeenCalledWith(instance.props.handleFiltersSubmit);
    expect(instance.props.handleFiltersSubmit).toHaveBeenCalled();
  });

  it('tests normalize property function', () => {
    expect(searchBar.props().normalize('John Doe')).toEqual('%John Doe%');
    expect(searchBar.props().normalize('John')).toEqual('%John%');
    expect(searchBar.props().normalize('')).toEqual('%%');
  });

  it('tests format property function', () => {
    expect(searchBar.props().format('%John Doe%')).toEqual('John Doe');
    expect(searchBar.props().format('%John%')).toEqual('John');
    expect(searchBar.props().format('%%')).toEqual('');
  });
});
