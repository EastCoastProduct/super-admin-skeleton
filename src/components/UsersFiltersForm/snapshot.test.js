import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheet, StyleSheetTestUtils } from 'aphrodite/no-important';
import UsersFiltersForm from './';

jest.mock('redux-form/immutable', () => ({
  Field: 'Field',
  reduxForm: () => component => component,
}));

describe('Pagination component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders required data', () => {
    const tree = renderer.create(
      <UsersFiltersForm
        form="UsersFilters"
        handleFiltersSubmit={() => {}}
        handleSubmit={f => f => f}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders required data with style', () => {
    const style = StyleSheet.create({
      filters: {
        margin: 0,
        padding: 0,
      },
    });
    const tree = renderer.create(
      <UsersFiltersForm
        className={style.filters}
        form="UsersFilters"
        handleFiltersSubmit={() => {}}
        handleSubmit={f => f => f}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
