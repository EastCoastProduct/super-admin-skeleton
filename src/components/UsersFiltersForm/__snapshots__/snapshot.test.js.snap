exports[`Pagination component snapshot renders required data 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={[Function]}>
  <Field
    className={
      Object {
        "_definition": Object {
          "display": "inline-block",
          "marginRight": 40,
          "verticalAlign": "middle",
        },
        "_name": "checkbox_6v2je1",
      }
    }
    component={[Function]}
    id="UsersFilters"
    label="Show only confirmed users"
    name="confirmed"
    onChange={[Function]} />
  <Field
    className={
      Object {
        "_definition": Object {
          "display": "inline-block",
          "verticalAlign": "middle",
          "width": 350,
        },
        "_name": "searchBar_ws3tt2",
      }
    }
    component={[Function]}
    format={[Function]}
    name="search"
    normalize={[Function]}
    placeholder="Search by email, first and last name"
    type="text" />
</form>
`;

exports[`Pagination component snapshot renders required data with style 1`] = `
<form
  className="filters_1433df3"
  noValidate={true}
  onSubmit={[Function]}>
  <Field
    className={
      Object {
        "_definition": Object {
          "display": "inline-block",
          "marginRight": 40,
          "verticalAlign": "middle",
        },
        "_name": "checkbox_6v2je1",
      }
    }
    component={[Function]}
    id="UsersFilters"
    label="Show only confirmed users"
    name="confirmed"
    onChange={[Function]} />
  <Field
    className={
      Object {
        "_definition": Object {
          "display": "inline-block",
          "verticalAlign": "middle",
          "width": 350,
        },
        "_name": "searchBar_ws3tt2",
      }
    }
    component={[Function]}
    format={[Function]}
    name="search"
    normalize={[Function]}
    placeholder="Search by email, first and last name"
    type="text" />
</form>
`;
